$OpenBSD: patch-src_winwidget_c,v 1.7 2011/07/22 08:04:31 dcoppa Exp $

Respect --image-bg option in full-screen mode
(upstream git commit 0379af967bddd6e90e4c7dadee6d96be1cb82f1d)

--- src/winwidget.c.orig	Mon Jul  4 14:49:17 2011
+++ src/winwidget.c	Fri Jul 22 09:59:28 2011
@@ -350,7 +350,10 @@ void winwidget_setup_pixmaps(winwidget winwid)
 			if (winwid->gc == None) {
 				XGCValues gcval;
 
-				gcval.foreground = BlackPixel(disp, DefaultScreen(disp));
+				if (!strcmp(opt.image_bg, "white"))
+					gcval.foreground = WhitePixel(disp, DefaultScreen(disp));
+				else
+					gcval.foreground = BlackPixel(disp, DefaultScreen(disp));
 				winwid->gc = XCreateGC(disp, winwid->win, GCForeground, &gcval);
 			}
 			winwid->bg_pmap = XCreatePixmap(disp, winwid->win, scr->width, scr->height, depth);
